### ISSUE TITLE
Override default extension functionality

### ISSUE THREAD

**Issue by yess-webioza** [nittro/nittro#33]
Hi, is there way to override a default extension functionality? For example if i want render & handle flash messages using external library (pnotify). 

I need to override `Nittro.Page.Bridges.PageFlashes.FlashAgent` and prevent default behaviour.

**Comment by jahudka** [comment]
Hi, well, yes, the easiest way to override an extension's functionality is to not include the extension in your build in the first place and provide a different implementation instead ;-) specifically with flashes you don't need anything from the nittro-flashes package if you have a custom renderer like pnotify - you just need a bridge to connect pnotify to the flashes received in ajax requests' payload, which is what a transaction agent is for (and it's exactly what `Nittro.Page.Bridges.PageFlashes.FlashAgent` is for). Let me know if you need any help implementing it, but I think that from reading the `FlashAgent`'s code it should be pretty clear what you need to do to get the flashes from the payload.

**Comment by Yess11** [comment]
Oh, that's true .. there should be no problem with implementation. Thanks anyway. 

And i also figgured out how to override that ( alternative solution ). Found `_context.mixin` function .. which is exactly what i was looking for:

```js

_context.invoke('App.Overrides', function (FlashAgent) {

    var FlasAgentMixin = {
        _showFlashes: function (flashes) {
            var id, i;

            for (id in flashes) {
                if (flashes.hasOwnProperty(id) && flashes[id]) {
                    for (i = 0; i < flashes[id].length; i++) {
                        console.log(flashes[id][i].message); // do what u want
                    }
                }
            }
        }
    };


    _context.mixin(FlashAgent, FlasAgentMixin);

}, {
    FlashAgent: 'Nittro.Page.Bridges.PageFlashes.FlashAgent'
});

**Comment by jahudka** [comment]
Yeah that'll work too - although I wouldn't recommend it as the 'cleanest' solution for two reasons:

1. As I said the `FlashAgent` will only be included in Nittro builds containing both `nittro-page` (which defines it) *and* `nittro-flashes` - but since you're effectively not using any functionality from `nittro-flashes` it seems like a waste of kilobytes to include it in the build. The full code for a custom transaction agent is something like 5 longer than what you already have.
2. It feels semantically weird to use a mixin - the mixin pattern is usually used for *augmenting* rather than *overriding* a class - but that's just a nitpick. Just for the sake of completeness though, your implementation could just directly assign your custom function to `FlashAgent.prototype._showFlashes` and the effect would be exactly the same :-)
